Author: Andreas Beckmann <anbe@debian.org>
Description: rename the ARCH variable
 the NVIDIA build systems reacts very fragile if this contains confusing
 values from different sources

--- a/Makefile
+++ b/Makefile
@@ -52,8 +52,8 @@ else
   LD ?= ld
   OBJDUMP ?= objdump
 
-  ifndef ARCH
-    ARCH := $(shell uname -m | sed -e 's/i.86/i386/' \
+  ifndef nvkARCH
+    nvkARCH := $(shell uname -m | sed -e 's/i.86/i386/' \
       -e 's/armv[0-7]\w\+/arm/' \
       -e 's/aarch64/arm64/' \
       -e 's/ppc64le/powerpc/' \
@@ -70,7 +70,7 @@ else
     KBUILD_PARAMS += V=1
   endif
   KBUILD_PARAMS += -C $(KERNEL_SOURCES) M=$(CURDIR)
-  KBUILD_PARAMS += ARCH=$(ARCH)
+  KBUILD_PARAMS += ARCH=$(nvkARCH)
   KBUILD_PARAMS += NV_KERNEL_SOURCES=$(KERNEL_SOURCES)
   KBUILD_PARAMS += NV_KERNEL_OUTPUT=$(KERNEL_OUTPUT)
   KBUILD_PARAMS += NV_KERNEL_MODULES="$(NV_KERNEL_MODULES)"
@@ -100,7 +100,7 @@ else
   # and hence must be used whenever present.
 
   LD_SCRIPT ?= $(KERNEL_SOURCES)/scripts/module-common.lds      \
-               $(KERNEL_SOURCES)/arch/$(ARCH)/kernel/module.lds \
+               $(KERNEL_SOURCES)/arch/$(nvkARCH)/kernel/module.lds \
                $(KERNEL_OUTPUT)/scripts/module.lds
   NV_MODULE_COMMON_SCRIPTS := $(foreach s, $(wildcard $(LD_SCRIPT)), -T $(s))
 
